### PR TITLE
chore(mme): new unit tests for mme procedure procedure failures, idle mode, etc.

### PR DESCRIPTION
**Pull request by ulaskozat**
## Summary

- Added new unit tests for authentication request and SMC message transmission failures.
- Added unit test for attach-idle mode transition-detach sequence.
- Added modify bearer request stages where it is needed.
- Expanded expect call macro with additional fields and added new message types as boundary conditions that wakes up the main thread waiting on the conditional variable.
## Test Plan

Run `make test_oai`.
## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.cpp [hunk 1]
```diff
@@ -244,5 +244,55 @@ void send_ue_capabilities_ind() {
...
+  S1AP_UE_CONTEXT_RELEASE_REQ(message_p).mme_ue_s1ap_id = 1;
+  S1AP_UE_CONTEXT_RELEASE_REQ(message_p).enb_ue_s1ap_id = 0;
+  S1AP_UE_CONTEXT_RELEASE_REQ(message_p).enb_id         = 0;
+  S1AP_UE_CONTEXT_RELEASE_REQ(message_p).relCause       = rel_cause;
+  send_msg_to_task(&task_zmq_ctx_main, TASK_MME_APP, message_p);
+  return;
+}
+
+void send_modify_bearer_resp(
+    std::vector<int>& bearer_to_modify, std::vector<int>& bearer_to_remove) {
```
Comment (ardzoht) [inline lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.cpp]:
```suggestion
    const std::vector<int>& bearer_to_modify, const std::vector<int>& bearer_to_remove) {
```

File: lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.cpp [hunk 2]
```diff
@@ -244,5 +244,55 @@ void send_ue_capabilities_ind() {
...
+    modify_response_p->bearer_contexts_modified.bearer_contexts[i]
+        .eps_bearer_id = bearer_to_modify[i];
+    modify_response_p->bearer_contexts_modified.bearer_contexts[i]
+        .cause.cause_value = REQUEST_ACCEPTED;
+  }
+  modify_response_p->bearer_contexts_modified.num_bearer_context =
+      bearer_to_modify.size();
+  for (int i = 0; i < bearer_to_modify.size(); ++i) {
+    modify_response_p->bearer_contexts_marked_for_removal.bearer_contexts[i]
+        .eps_bearer_id = bearer_to_modify[i];
```
Comment (ardzoht) [inline lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.cpp]: this should be `bearer_to_remove[i]`

File: lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.h [hunk 3]
```diff
@@ -72,5 +82,12 @@ void send_ue_ctx_release_complete();
 
 void send_ue_capabilities_ind();
 
+void send_context_release_req(s1cause rel_cause, task_id_t TASK_ID);
+
+void send_modify_bearer_resp(
+    std::vector<int>& bearer_to_modify, std::vector<int>& bearer_to_remove);
```
Comment (ardzoht) [inline lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.h]:
```suggestion
    const std::vector<int>& bearer_to_modify, const std::vector<int>& bearer_to_remove);
```

**Review by ardzoht — CHANGES_REQUESTED** [review]
minor changes on send_modify_bearer_resp utility function

File: lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.cpp (hunk 2 above)
Comment (ulaskozat) [reply lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.cpp]: good catch!

**Review by ardzoht — APPROVED** [review]
lgtm
